src/cryptography/hazmat/primitives/kdf/kbkdf.py

```diff
@@ -117,14 +135,19 @@ def derive(self, key_material: bytes, prf_output_size: int) -> bytes:
         for i in range(1, rounds + 1):
             h = self._prf(key_material)
 
+            fixed = self._generate_fixed_input()
             counter = utils.int_to_bytes(i, self._rlen)
+
             if self._location == CounterLocation.BeforeFixed:
                 h.update(counter)
-
-            h.update(self._generate_fixed_input())
-
-            if self._location == CounterLocation.AfterFixed:
+                h.update(fixed)
+            elif self._location == CounterLocation.AfterFixed:
+                h.update(fixed)
+                h.update(counter)
+            else:
+                h.update(fixed[: self._break_location])
```

Comment:
I think we should validate break location, if it's negative or `> len(fixed)` dumb things happen.

Reply:
OK, will do.

Reply:
Updated. Quite a bit had to change, might be best to review the MR in its entirety again... :cry: